test/v2/controllers/AmendPeriodSummaryControllerSpec.scala

```diff
@@ -39,40 +37,47 @@ import scala.concurrent.Future
 
 class AmendPeriodSummaryControllerSpec
     extends ControllerBaseSpec
-    with MockEnrolmentsAuthService
-    with MockMtdIdLookupService
+    with ControllerTestRunner
     with MockAmendPeriodSummaryService
     with MockAmendPeriodSummaryRequestParser
     with MockHateoasFactory
-    with MockIdGenerator
     with AmendPeriodSummaryFixture {
 
-  private val nino          = "AA123456A"
-  private val businessId    = "XAIS12345678910"
-  private val periodId      = "2019-01-01_2020-01-01"
-  private val correlationId = "X-123"
-  private val taxYear       = "2023-24"
-
-  val testHateoasLinks: Seq[Link] =
-    Seq(Link(href = "/some/link", method = GET, rel = "someRel"))
-
-  trait Test {
-    val hc: HeaderCarrier = HeaderCarrier()
-
-    val controller = new AmendPeriodSummaryController(
-      authService = mockEnrolmentsAuthService,
-      lookupService = mockMtdIdLookupService,
-      parser = mockAmendPeriodSummaryRequestParser,
-      service = mockAmendPeriodSummaryService,
-      hateoasFactory = mockHateoasFactory,
-      cc = cc,
-      idGenerator = mockIdGenerator
+  private val businessId: String  = "XAIS12345678910"
+  private val periodId: String    = "2019-01-01_2020-01-01"
+  private val tysPeriodId: String = "2024-01-01_2025-01-01"
+  private val taxYear: String     = "2023-24"
+
+  val testHateoasLinks: Seq[Link] = Seq(
+    Link(
+      href = s"/individuals/business/self-employment/$nino/$businessId/period/$periodId",
+      method = PUT,
+      rel = "amend-self-employment-period-summary"
+    ),
+    Link(href = s"/individuals/business/self-employment/$nino/$businessId/period/$periodId", method = GET, rel = "self"),
+    Link(
+      href = s"/individuals/business/self-employment/$nino/$businessId/period",
+      method = GET,
+      rel = "list-self-employment-period-summaries"
     )
-
-    MockMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
-    MockEnrolmentsAuthService.authoriseUser()
-    MockIdGenerator.generateCorrelationId.returns(correlationId)
-  }
+  )
+
+  private val testTysHateoasLink = Seq(
+    hateoas.Link(
```

Comment:
we can remove the `hateoas.` from the start of each of these and just have `Link(...)`